src/main/java/org/spongepowered/common/profile/SpongeGameProfileManager.java

```diff
@@ -171,12 +182,32 @@ public void lookupUserAsync(final UUID uniqueId) {
             } catch (final InterruptedException | ExecutionException e) {
                 e.printStackTrace();
             }
-
-            try {
-                Thread.sleep(SpongeConfigs.getCommon().get().world.gameProfileQueryTaskInterval * 1000L);
-            } catch (final InterruptedException e) {
-                e.printStackTrace();
+            if (SpongeGameProfileManager.canLookup(uniqueId)) { // only wait when it's possible that a lookup has actually occurred
+                try {
+                    Thread.sleep(SpongeConfigs.getCommon().get().world.gameProfileQueryTaskInterval * 1000L);
+                } catch (final InterruptedException e) {
+                    e.printStackTrace();
+                }
             }
         });
     }
+
+    public void close() {
+        // Shut down the executor
+        this.gameLookupExecutorService.shutdown();
+
+        try {
+            if (!this.gameLookupExecutorService.awaitTermination(10, TimeUnit.SECONDS)) {
```

Comment:
I wonder if this may be too high? Maybe 5 is more reasonable unless this value was sourced from elsewhere.

Reply:
sure? It's a bit of a block i've copied around a few projects, haven't thought about the time in a while